utils/Azure/models/image.ts

```diff
@@ -17,14 +17,18 @@ export class AzureImage {
     public password?: string;
     public username?: string;
 
-    constructor(repository: Repository, tag: string, created: Date) {
+    constructor(repository: Repository, tag: string, created?: Date) {
         this.registry = repository.registry;
         this.repository = repository;
         this.tag = tag;
-        this.created = created;
         this.subscription = repository.subscription;
         this.resourceGroupName = repository.resourceGroupName;
+        if (created) { this.created = created; }
```

Comment:
"this.created = created" is safe

Reply:
even if created is an optional parameter?

Reply:
Yes. Optional here is the same as "Date | undefined".  So if not passed in, you'll just assign undefined to this.created, which is exactly what you'd get if queried if before it's set to anything.

Technically, there have been a few cases where this would make a difference (setting it to undefined actually creates an "own" property on the object with the value undefined, vs there not being a property with that name on the object at all), I've seen some API functions that responded differently. But most well-written code won't care about the difference.